src/scripts/generator_base.py

```diff
@@ -0,0 +1,86 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import argparse
+import random
+import math
+import sys
+
+# Local packages
+import settings
+path = os.path.join(settings.TEMPLATE_DIR)
+sys.path.append(path)
+import validate # Function
+from constants import _constants
+
+# Generator base class, other generator classes will intherit from this class
+class Generator:
+    test_case = "" # Actual test case with validate function call
+    file_output = [] # All the test cases generated (without function definition)
+
+    # Function for printing debug messages if the user wants to (-q switch : quiet)
+    def debug(self, message, options):
+        if not options.q:
+            print(message)
+
+    # Function for generating a random false value integer
+    def random_false_int(self, expected_value, MIN, MAX):
+        # Random number wich must not be equal with the expected value
+        if ((random.randint(0, 1) and expected_value != MAX) or expected_value == MIN):
+            N1 = random.randint(int(expected_value + 1), MAX)
+        else:
+            N1 = random.randint(MIN, int(expected_value - 1))
+        return N1
+
+    # Generating false numbers and returning with an array
+    def generate_false_numbers(self, expected_value, false_values_count, MIN, MAX):
+        false_values = []
+        for j in range(false_values_count):
+            false_values.append(self.random_false_int(expected_value, MIN, MAX))
+        return false_values
+
+    # Creating a string wich contains an array of false values with variable names, only no. of false values are needed
+    def create_false_numbers_array_string(self, false_values_count):
+        false_values = "["
+        for i in range(false_values_count):
+            false_values += "%s" % (_constants.FALSE_RESULT_VAR_NAME.format(NUMBER = i + 1))
+            if (i != false_values_count - 1):
+                false_values += ", "
+        false_values += "]"
+        return false_values
+
+    # Create a string for creating declarations of the values in the array each as a variable
+    # Or the declaration of the value
+    def create_declarations(self, values, var_name):
+        declaration = ""
+        if isinstance(values, list):
+            for i in range(len(values)): # For eg: a0 = 38473837;
+                declaration += var_name.format(NUMBER = i + 1)
+                declaration += " = %s;\n" % (str(values[i]))
+        else:
+            declaration += var_name.format(NUMBER = "")
+            declaration += " = %s;\n" % (str(values))
+        return declaration
```

Comment:
This kind of similar to the function above. Please try to apply the suggestion to this function as well.

Reply:
Is it okay like this?
```
    def create_declarations(self, values, var_name):
        declaration = ""
        if isinstance(values, list):
            for i in range(1, len(values) + 1): # For eg: a0 = 38473837;
                 declaration += "%s = %s;\n" % (var_name.format(NUMBER = i), str(values[i - 1]))
        else: # If we have only one value no number added to the variable name, for eg: a = 38473837;
            declaration = "%s = %s;\n" % (var_name.format(NUMBER = ""), (str(values)))
        return declaration
```
I could'nt manage to do it that way, and I think in this case it is more readable this way. The code became 2 lines shorter even by this. But in the other case it was useful.

Reply:
Better! But in this function we create a "list" from declarations, and store it in a concatenated string. How about storing the list instead? e.g { "f1" : "5", "f2" : "10, ... }